substrate/frame/staking/src/lib.rs

```diff
@@ -1081,69 +1074,54 @@ impl<T: Config> Convert<T::AccountId, Option<Exposure<T::AccountId, BalanceOf<T>
-pub type Existence = ();
-
-/// A converter type that returns `Some(())` if the validator exists in the current active era,
-/// otherwise `None`. This serves as a lightweight presence check for validators.
-pub struct ExistenceOf<T>(core::marker::PhantomData<T>);
-impl<T: Config> Convert<T::AccountId, Option<Existence>> for ExistenceOf<T> {
-	fn convert(validator: T::AccountId) -> Option<Existence> {
-		Validators::<T>::contains_key(&validator).then_some(())
-	}
-}
-
-/// A compatibility wrapper type used to represent the presence of a validator in the current era.
-/// Encodes as type [`Existence`] but can decode from legacy [`Exposure`] values for backward
-/// compatibility.
-#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, RuntimeDebug, TypeInfo, DecodeWithMemTracking)]
-pub enum ExistenceOrLegacyExposure<A, B: HasCompact> {
-	/// Validator exists in the current era.
-	Exists,
-	/// Legacy `Exposure` data, retained for decoding compatibility.
-	Exposure(Exposure<A, B>),
-}
-
-/// Converts a validator account ID to a Some([`ExistenceOrLegacyExposure::Exists`]) if the
-/// validator exists in the current era, otherwise `None`.
-pub struct ExistenceOrLegacyExposureOf<T>(core::marker::PhantomData<T>);
-
-impl<T: Config> Convert<T::AccountId, Option<ExistenceOrLegacyExposure<T::AccountId, BalanceOf<T>>>>
-	for ExistenceOrLegacyExposureOf<T>
-{
-	fn convert(
-		validator: T::AccountId,
-	) -> Option<ExistenceOrLegacyExposure<T::AccountId, BalanceOf<T>>> {
-		ActiveEra::<T>::get()
-			.map(|active_era| ErasStakersOverview::<T>::contains_key(active_era.index, &validator))
-			.unwrap_or(false)
-			.then_some(ExistenceOrLegacyExposure::Exists)
-	}
-}
+/// Identify a validator with their default exposure.
+///
+/// This type should not be used in a fresh runtime, instead use [`UnitIdentificationOf`].
+///
+/// In the past, a type called [`ExposureOf`] used to return the full exposure of a validator to
+/// identify their exposure. This type is kept, marked as deprecated, for backwards compatibility of
+/// external SDK users, but is no longer used in this repo.
+///
+/// In the new model, we don't need to identify a validator with their full exposure anymore, and
+/// therefore [`UnitIdentificationOf`] is perfectly fine. Yet, for runtimes that used to work with
+/// [`ExposureOf`], we need to be able to decode old identification data, possibly stored in the
+/// historical session pallet in older blocks. Therefore, this type is a good compromise, allowing
+/// old exposure identifications to be decoded, and returning a few zero bytes
+/// (`Exposure::default`) for any new identification request.
+///
+/// A typical usage of this type is:
+///
+/// ```ignore
+/// impl pallet_session::historical::Config for Runtime {
+///     type FullIdentification = sp_staking::Exposure<AccountId, Balance>;
+///     type IdentificationOf = pallet_staking::DefaultExposureOf<Self>
+/// }
+/// ```
+pub struct DefaultExposureOf<T>(core::marker::PhantomData<T>);
```

Comment:
Should we move this to AH Client? Then it plays well with why we need to use `DefaultExposure`.

Reply:
A copy of it is indeed there too ;) 

as we remove pallet-staking from everywhere, that copy will be used.